### PR TITLE
[Clp] New recipe. v1.16.11

### PR DESCRIPTION
**Pull request by ViralBShah**
Following https://github.com/JuliaOpt/ClpBuilder

Currently disables mumps. We'll enable it once the library is building.

### CODE REVIEW

**Comment by ViralBShah** [comment]
How do I depend on METIS@4 in my dependencies? I am using `METIS_jll@4`, which works when I do `Pkg.add()`, but BB does not like it.

https://dev.azure.com/JuliaPackaging/Yggdrasil/_build/results?buildId=2480&view=logs&j=be5a51e8-fadf-5b1a-9ea5-0b8a2abebf40&t=91f4752b-623f-5b7d-cd90-5fb74adeae1b&l=15

File: C/Clp/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    LibraryProduct("libClp", :libClp),
+    LibraryProduct("libClpSolver", :libClpSolver)
+]
+
+# Dependencies that must be installed before this package can be built
+dependencies = [
+    Dependency("CoinUtils_jll"),
+    Dependency("Osi_jll"),
+    Dependency("ASL_jll"),
+    Dependency("METIS_jll@4"),
```
Comment (giordano) [inline C/Clp/build_tarballs.jl:50]: Probably here you want to do
```suggestion
    Dependency(PackageSpec(; name = "METIS_jll", uuid = "d00139f3-1899-568f-a2f0-47f597d42d70", version = v"4.0.3")),
```

You need also to load `Pkg` at the top

**Comment by giordano** [comment]
Wow, it looks like each platforms fails for a different reason almost, this is super fun.  Note that some dependencies are not available for all platforms.  `x86_64-linux-gnu` fails because it looks like
```
01:06:11] c++ -O3 -pipe -DNDEBUG -Wparentheses -Wreturn-type -Wcast-qual -Wall -Wpointer-arith -Wwrite-strings -Wconversion -Wno-unknown-pragmas -Wno-long-long -DCLP_BUILD -o .libs/clp ClpMain.o  ./.libs/libClpSolver.so /workspace/srcdir/Clp/build/Clp/src/.libs/libClp.so ./.libs/libClp.so /opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/6.1.0/../../../../x86_64-linux-gnu/lib/../lib64//libstdc++.so /workspace/destdir/lib/libCoinUtils.so -lopenblas -lasl -Wl,--rpath -Wl,/workspace/destdir/lib -Wl,--rpath -Wl,/opt/x86_64-linux-gnu/bin/../lib/gcc/x86_64-linux-gnu/6.1.0/../../../../x86_64-linux-gnu/lib/../lib64/
[01:06:11] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/lib/libasl.so: undefined reference to `dlopen'
[01:06:11] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/lib/libasl.so: undefined reference to `dlclose'
[01:06:11] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/lib/libasl.so: undefined reference to `clock_gettime'
[01:06:11] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/lib/libasl.so: undefined reference to `dlerror'
[01:06:11] /opt/x86_64-linux-gnu/bin/../x86_64-linux-gnu/sys-root/usr/local/lib/libasl.so: undefined reference to `dlsym'
[01:06:11] collect2: error: ld returned 1 exit status
[01:06:11] make[3]: *** [Makefile:814: clp] Error 1
```
should be linked to `libdl` with `-ldl`.

**Comment by ViralBShah** [comment]
Most of the builds are failing due to issue with `libstdc++.la`.

**Comment by ViralBShah** [comment]
@giordano When you have ample time, can you look at each of the failures and share your wisdom? :-)

Each one is failing in its own way.

**Comment by ViralBShah** [comment]
The main failure across many platforms seems to be:
```
[19:01:08] grep: /workspace/destdir/aarch64-linux-gnu/lib/../lib64/libstdc++.la: No such file or directory
[19:01:08] sed: /workspace/destdir/aarch64-linux-gnu/lib/../lib64/libstdc++.la: No such file or directory
[19:01:08] libtool: link: `/workspace/destdir/aarch64-linux-gnu/lib/../lib64/libstdc++.la' is not a valid libtool archive
```

Some googling suggests this is due to autoconf scripts not setup properly for cross-compilation. Does anyone have an understanding of this and how I could possibly coerce it into building? For reference, the current ClpBuilder is here. It bundles all the dependencies although I feel like unbundling should work, but that is not the problem right now.

https://github.com/JuliaOpt/ClpBuilder

**Comment by giordano** [comment]
Try removing all libtool files:
```
rm -f ${prefix}/lib/*.la
```
We have some problems with them, they're wrong

**Comment by ViralBShah** [comment]
I am getting link errors like these. The Osi library uses `expand_cxxstring_abis`. Do I have to do the same expansion here as well?

```
[00:23:08] c++ -shared -nostdlib /opt/i686-w64-mingw32/bin/../i686-w64-mingw32/sys-root/lib/../lib/dllcrt2.o /opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/crtbegin.o  .libs/OsiClpSolverInterface.o  -L/workspace/destdir/bin -lOsi ../.libs/libClp.dll.a -L/opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/../../../../i686-w64-mingw32/lib/../lib/ -L/opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0 -L/opt/i686-w64-mingw32/bin/../lib/gcc -L/opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/../../../../i686-w64-mingw32/lib/../lib -L/opt/i686-w64-mingw32/bin/../i686-w64-mingw32/sys-root/lib/../lib -L/opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/../../../../i686-w64-mingw32/lib -L/opt/i686-w64-mingw32/bin/../i686-w64-mingw32/sys-root/lib /opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/../../../../i686-w64-mingw32/lib/../lib//libstdc++.dll.a -lmingw32 -lgcc_s -lgcc -lmoldname -lmingwex -lmsvcrt -lpthread -ladvapi32 -lshell32 -luser32 -lkernel32 -lmingw32 -lgcc_s -lgcc -lmoldname -lmingwex -lmsvcrt /opt/i686-w64-mingw32/bin/../lib/gcc/i686-w64-mingw32/6.1.0/crtend.o  -o .libs/libOsiClp-1.dll -Wl,--enable-auto-image-base -Xlinker --out-implib -Xlinker .libs/libOsiClp.dll.a
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0xa27): undefined reference to `CoinWarmStartBasis::CoinWarmStartBasis()'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0xc1f): undefined reference to `CoinWarmStartBasis::CoinWarmStartBasis(CoinWarmStartBasis const&)'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x1092): undefined reference to `CoinPackedVector::setFullNonZero(int, double const*, bool)'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x1149): undefined reference to `CoinIndexedVector::clear()'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x1152): undefined reference to `CoinIndexedVector::clear()'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x115b): undefined reference to `CoinIndexedVector::clear()'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x1162): undefined reference to `CoinIndexedVector::clear()'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x11ac): undefined reference to `CoinIndexedVector::insert(int, double)'
[00:23:08] .libs/OsiClpSolverInterface.o:OsiClpSolverInterface.cpp:(.text+0x12ac): undefined reference to `CoinIndexedVector::insert(int, double)'
```

**Comment by ViralBShah** [comment]
I think there is no PR opened up on General.

**Pull request merged**